Update docs for ISP and PRS (#930)
vitessio/website: content/en/docs/13.0/user-guides/configuration-basic/initialize-shard-primary.md

---
title: Initialize Shard Primary
weight: 9
---

> The [PlannedReparentShard](../../configuration-advanced/reparenting/#plannedreparentshard-planned-reparenting) should be used to initialize the shard by electing a new primary and setting up replication for the replicas. Additionally, a database is created to store the data for the keyspace-shard. The default name for the database will be `vt_` followed by the keyspace. In our use case, it would be `vt_commerce`. You can override this default by providing an `init_db_name_override` flag to vttablet. However, all future invocations must continue to supply this parameter.

Reviewer: ```suggestion
The [PlannedReparentShard](../../configuration-advanced/reparenting/#plannedreparentshard-planned-reparenting) command should be used to initialize the shard by electing a new primary and setting up replication for the replicas. Additionally, a database is created to store the data for the keyspace-shard. The default name for the database will be `vt_` followed by the keyspace. In our use case, it would be `vt_commerce`. You can override this default by providing an `init_db_name_override` flag to each vttablet. However, all future invocations must continue to supply this parameter.
```

Author: Fixed

The InitShardPrimary step can also be used to do the same operation. However, it is a destructive command and should only be used by advanced users. This command copies over the `executed_gtid_set` from the primary to the replica which can break replication if the user isn't careful. 

The command for `InitShardPrimary` is as follows:

```text
vtctlclient \
  InitShardPrimary \
  -force \
  commerce/0 \
  cell1-100
```

> Until one of these command are run, you may also see errors like this in the vttablet logs: `Cannot start query service: Unknown database 'vt_xxx'`. This is because the database will be created only after a primary is elected.

Reviewer: ```suggestion
Until one of these commands is run, you may also see errors like this in the vttablet logs: `Cannot start query service: Unknown database 'vt_xxx'`. This is because the database will be created only after a primary is elected.
```

Author: Fixed

> If you have semi-sync enabled and did not set up at least three replicas, InitiShardPrimary could hang indefinitely. Even if it succeeds, future operations that perform failovers could cause this shard to go into a deadlocked state.

Reviewer: ```suggestion
If you have semi-sync enabled and did not set up at least three replicas, InitShardPrimary could hang indefinitely. Even if it succeeds, future operations that perform failovers could cause this shard to go into a deadlocked state.
```

Reviewer: We only need two replicas, not three?

Author: I think we meant 3 total tablets... I am changing it to two replicas now

After this step, visiting the `/debug/status` page on the vttablets should show all the tablets as healthy:

![healthy-tablet](../img/healthy-tablet.png)

{{< warning >}}
`InitShardPrimary` is a destructive command that resets all servers by deleting their binlog metadata. It should only be used for initializing a brand new cluster.
{{< /warning >}}

{{< info >}}

> `InitShardPrimary` will soon be deprecated. This action will be performed automatically by `vtorc` once it is released as production-ready.

Reviewer: Haven't we deprecated it already?

Author: Have we? Is there some process we need to follow to do that?

Reviewer: ok, we haven't deprecated it yet. Once that is done we can update the docs appropriately.
From vtctlclient help:
```
  InitShardPrimary [-force] [-wait_replicas_timeout=<duration>] <keyspace/shard> <tablet alias>
```

Reviewer: Example of how to deprecate can be seen by looking at `InitTablet`.

Author: Alright! Thankyou 🙇‍♂️
